src/mudlet-lua/lua/Other.lua

```diff
@@ -846,3 +846,69 @@ do
     end
   end
 end
+
+local timeframetable = {}
+
+function timeframe(vname, true_time, nil_time, ...)
+  local format = string.format
+
+  assert(type(true_time) == "number" or type(true_time) == "table", format("timeframe: bad argument #2 type (true time as a number or table expected, got %s!", type(true_time)))
+  assert(type(nil_time) == "nil" or type(nil_time) == "number" or type(nil_time) == "table", format("timeframe: bad argument #3 type (nil time as a number or table expected, got %s!", type(nil_time)))
+
+  -- aggregate timerlist data
+  local timerlist = {
+    {0, nil},
+    type(true_time) == "number" and {true_time, true} or type(true_time) == "table" and true_time,
+    type(nil_time) == "number" and {nil_time, nil} or type(nil_time) == "table" and nil_time,
+    ...
+  }
+
+  -- reinitialise timeframe for vname
+  killtimeframe(vname)
+  timeframetable[vname] = {}
+
+  local vtype = type(vname)
+
+  -- loop through timerlist and create tempTimers
+  local maxtime = 0
+  for step, data in ipairs(timerlist) do
+    assert(type(data) == "table", format("timeframe: bad argument #4 type (timerlist data as a table expected, got %s!", type(data)))
```

Comment:
We should not hardcode the argument number here. Instead, use `step - 2` if `nil_time` is a number, `step - 1` if `true_time` is a number and `step` if neither is a number.

Reply:
Could you elaborate on this, please?

Reply:
Doesn't the location of the timerlist in the function arguments change?

Reply:
The following calls would all result in an error message of `timeframe: bad argument #4 type (timerlist data as a table expected, got function!`, but the argument number is not right:
```lua
timeframe("foo", 5, {7.5, "bar"}, {10, "baz"}, function() end) -- we'd want to have #5 here
timeframe("foo", {5, "foo"}, {7.5, "bar"}, {10, "baz"}, {15, "foozle"}, function() end) -- we'd want to have #6 here
timeframe("foo", 5, 25, {7.5, "bar"}, {10, "baz"}, {15, "foozle"}, function() end) -- we'd want to have #7 here
```

The second sentence is complete bullshit and you'd better disregard that one. You only need to use `step` as number argument, if I'm not mistaken again.